redis/cache/redisx/request.go

```diff
@@ -261,9 +262,13 @@ func (r Request) convertAndSetByteSlice(dst reflect.Value, src reflect.Value) er
 	case reflect.String:
 		dst.Set(reflect.ValueOf(asStr))
 	default:
-		return &ResponseInputTypeError{
-			Cmd:               r.Cmd,
-			ResponseInputType: dst.Type(),
+		if dst.Type() == stringPtrT {
+			dst.Set(reflect.ValueOf(&asStr))
+		} else {
+			return &ResponseInputTypeError{
+				Cmd:               r.Cmd,
+				ResponseInputType: dst.Type(),
+			}
 		}
```

Comment:
🔕 this can be a switch too

```suggestion
		switch dst.Type() {
		case stringPtrT:
			dst.Set(reflect.ValueOf(&asStr))
		default:
			return &ResponseInputTypeError{
				Cmd:               r.Cmd,
				ResponseInputType: dst.Type(),
			}
		}
```